File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length())    return false;
        Map<Character,Integer> smap = new HashMap<>();
        Map<Character,Integer> tmap = new HashMap<>();
        for(int i=0; i<s.length(); i++){
            smap.put(s.charAt(i),smap.getOrDefault(s.charAt(i),0)+1);
            tmap.put(t.charAt(i),tmap.getOrDefault(t.charAt(i),0)+1);
        }
        return smap.equals(tmap);
    }
}

File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):  return False
        counter={}
        for char in s:
            counter[char] = counter.get(char,0)+1
        for char in t:
            if char not in counter or counter[char]==0:  return False
            counter[char]-=1
        return True